### PR TITLE
Dark mode

### LINKED ISSUE #56: Dark mode
**Is your feature request related to a problem? Please describe.**
In the newest iOS and Android version, you can enable a system-wide dark mode. Our app should adhere to that, choosing a darker theme.

**Describe the solution you'd like**
We should provide all TextStyles and colors through a `Theme` and provide a `darkTheme` to the `MaterialApp` to automatically adhere to the system setting.

### LINKED ISSUE #81: Migrate to AndroidX
**Is your feature request related to a problem? Please describe.**
Going forward, Android will develop the AndroidX libraries instead of Android Support Libraries.

**Describe the solution you'd like**
We should migrate the project to AndroidX.

### LINKED ISSUE #104: Update privacy policy link
**Describe the bug**
The privacy policy link is outdated. We should reference either https://schul-cloud.org/datenschutz or include a custom privacy policy.

### PR DESCRIPTION
**Pull request by JonasWanke**
**Fixes: #56, fixes: #81, fixes: #104**

## Theming for different instances

Individual apps are now available for:
- Schul-Cloud (sc)
- Schul-Cloud Brandenburg (brb)
- Niedersächsische Bildungscloud (n21)
- Open Schul-Cloud (open)
- Thüringer Schulcloud (thr)

`.vscode/launch.json` contains the correct start configurations for each of them.

| Schul-Cloud | Schul-Cloud Brandenburg | Niedersächsische Bildungscloud |
| :------------: | :------------: | :------------: |
| image | image | image |
| **Open Schul-Cloud** | **Thüringer Schulcloud** | **Launcher Icons** |
| image | image | image |

## Dark mode
(based on system settings)

| image | image | image |
| :------------: | :------------: | :------------: |
| image | image | image |
| image |  | |

## Miscellaneous

- update to **AndroidX**, update gradle version
- correct **privacy policy** link

### CODE REVIEW

**Comment by JonasWanke** [comment]
(Our CI fails as widgets don't properly provide diagnostics data. This is not related to this PR and is now being tracked in #109.)

**Comment by janrenz** [comment]
What happens with black logos like n21/nbc on the dark mode?

**Comment by janrenz** [comment]
We also need to figure out to get this running in iOS
see https://medium.com/@animeshjain/build-flavors-in-flutter-android-and-ios-with-different-firebase-projects-per-flavor-27c5c5dac10b

**Comment by JonasWanke** [comment]
@janrenz Unfortunately, I neither have any experience with developing iOS apps nor do I own any Apple-devices for testing these.

Regarding the logos: The contrast isn't always high enough for good readability on dark mode; I'll work on a fix.

**Comment by JonasWanke** [comment]
Some logos (n21, open, thr) now have a variation for use on dark backgrounds. These variations were created by replacing black with white text and using a lighter shade of dark logo colors. Also, some logos on the login screen are now bigger.

File: lib/app/theming/config.dart [hunk 1]
```diff
@@ -0,0 +1,139 @@
...
+    );
+  }
+}
+
+class _InheritedAppConfig extends InheritedWidget {
+  const _InheritedAppConfig({
+    Key key,
+    @required this.appConfig,
+    @required Widget child,
+  })  : assert(AppConfig != null),
```
Comment (MarcelGarus) [inline lib/app/theming/config.dart]: Probably should be lowecase `appConfig`. The type always exists.

File: lib/news/widgets/author.dart [hunk 2]
```diff
@@ -26,8 +26,7 @@ class _AuthorViewState extends State<AuthorView> {
   Widget _buildName() {
     return Container(
       alignment: Alignment.centerLeft,
-      child: Container(
-        color: Colors.white,
+      child: Padding(
```
Comment (MarcelGarus) [inline lib/news/widgets/author.dart]: Can't we just add the padding to the surrounding `Container`?

**Review by MarcelGarus — CHANGES_REQUESTED** [review]

**Review by MarcelGarus — APPROVED** [review]

**Comment by MarcelGarus** [comment]
Ready to merge!

**Comment by janrenz** [comment]
great PR!

**Pull request merged**